node/src/blockchain/blockchain_interface/mod.rs

```diff
@@ -0,0 +1,265 @@
+use web3::types::{Address, TransactionReceipt, H256};
+use web3::Error;
+
+pub trait BlockchainInterface {
+    fn contract_address(&self) -> Address;
+
+    fn retrieve_transactions(
+        &self,
+        start_block: u64,
+        recipient: &Wallet,
+    ) -> Result<RetrievedBlockchainTransactions, BlockchainError>;
+
+    fn build_blockchain_agent(
+        &self,
+        consuming_wallet: &Wallet,
+        persistent_config: &dyn PersistentConfiguration,
+    ) -> Result<Box<dyn BlockchainAgent>, String>;
+
+    fn send_batch_of_payables(
+        &self,
+        agent: Box<dyn BlockchainAgent>,
+        new_fingerprints_recipient: &Recipient<PendingPayableFingerprintSeeds>,
+        accounts: &[PayableAccount],
+    ) -> Result<Vec<ProcessedPayableFallible>, PayableTransactionError>;
+
+    fn get_transaction_receipt(&self, hash: H256) -> ResultForReceipt;
+
+    fn helpers(&self) -> &dyn RPCHelpers;
+}
+
+#[derive(Clone, Debug, Eq, Message, PartialEq)]
+pub struct BlockchainTransaction {
+    pub block_number: u64,
+    pub from: Wallet,
+    pub wei_amount: u128,
+}
+
+impl fmt::Display for BlockchainTransaction {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), fmt::Error> {
+        write!(
+            f,
+            "{}gw from {} ({})",
+            self.wei_amount, self.from, self.block_number
+        )
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum BlockchainError {
+    InvalidUrl,
+    InvalidAddress,
+    InvalidResponse,
+    QueryFailed(String),
+}
+
+impl Display for BlockchainError {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let err_spec = match self {
+            Self::InvalidUrl => Either::Left("Invalid url"),
+            Self::InvalidAddress => Either::Left("Invalid address"),
+            Self::InvalidResponse => Either::Left("Invalid response"),
+            Self::QueryFailed(msg) => Either::Right(format!("Query failed: {}", msg)),
+        };
+        write!(f, "Blockchain error: {}", err_spec)
+    }
+}
+
+pub type BlockchainResult<T> = Result<T, BlockchainError>;
+pub type ResultForReceipt = BlockchainResult<Option<TransactionReceipt>>;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PayableTransactionError {
+    MissingConsumingWallet,
+    GasPriceQueryFailed(String),
+    TransactionCount(BlockchainError),
+    UnusableWallet(String),
+    Signing(String),
+    Sending { msg: String, hashes: Vec<H256> },
+}
+
+impl Display for PayableTransactionError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::MissingConsumingWallet => {
+                write!(f, "Missing consuming wallet to pay payable from")
+            }
+            Self::GasPriceQueryFailed(msg) => {
+                write!(f, "Unsuccessful gas price query: \"{}\"", msg)
+            }
+            Self::TransactionCount(blockchain_err) => write!(
+                f,
+                "Transaction count fetching failed for: {}",
```

Comment:
How does that "for" read in the error message? I don't know what sort of messages you get in `blockchain_err`, but it seems as though it might be a little clumsy. It's a small nit to pick, but it might be better without the "for," or perhaps with something like "because of" or "due to" instead.